Fix Rust 1.80 cfg Linting and add VolumeRef impls

gierens opened the pull request:
Rust 1.80 which is used in functional tests does more extensive cfg linting.
Because `volume` and `identity` are not known it causes functional tests to fail.
This adds a linting rule to allow both those values.

For more info see: https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html#checked-cfg-names-and-values

gierens commented:
1.65 tests are failing due to tokio requiring rust 1.70, see #159 for fix

gierens commented:
Zed fails, but is EOL anyway, see #160

dtantsur commented:
@gierens hi! Shouldn't we add them to the feature list if we use them?

dtantsur commented:
Took a deeper look. I guess "volume" should be called "block-storage" now, so it's actually a bug. "identity" is reserved for the future, so maybe it's fair to exclude it in the linter, but also it may be better to just add it.

gierens commented:
> Took a deeper look. I guess "volume" should be called "block-storage" now, so it's actually a bug. "identity" is reserved for the future, so maybe it's fair to exclude it in the linter, but also it may be better to just add it.

@dtantsur Yeah, I didn't look too much into the code that caused the issue, but I was a little suspicious of those features as well.

I adjusted this accordingly:
- `identity` and `test` features were added with comments explaining why they are not used
- `volume` for `VolumeRef` was renamed to `block-storage`, and I also implemented the missing `impl`s for `VolumeRef`
- for `SnapshotRef` this is not going to work, though, since there is no snapshot API yet, so I added a temporary feature `block-storage-snapshot` with comments that this is going to be removed once `block-storage` has the snapshot API and can thus implement `SnapshotRef`. Adding the snapshot API is actually one of the things I'm planning to do, once the CI is working again, so this is really just a temporary fix.

gierens commented:
> Took a deeper look. I guess "volume" should be called "block-storage" now, so it's actually a bug. "identity" is reserved for the future, so maybe it's fair to exclude it in the linter, but also it may be better to just add it.

@dtantsur Yeah you're right, I should've looked into the code causing the issue ... I adjusted the PR accordingly:
- I added the `identity` feature with a comment that it's reserved for future use.
- The `VolumeRef` feature was changed to `block-storage` and I added the missing `impl`s in the `block-storage` module.
- For `SnapshotRef` this is not gonna work without a snapshot API though, so simply ignored it in the linting for now, but added TODO comments that this is going to get the `block-storage` feature once snapshots have been implemented. This is actually next on my list after the CI is fixed, so should really just be a temporary thing.
- The `test` feature which is used in some tests I also added to the linting rules, since this is not a real feature.

gierens commented:
@dtantsur Alright, CI seems to approve. So this is ready for another review.

dtantsur commented:
Thanks much!
